src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordHelpers.cs

```diff
@@ -266,74 +252,184 @@ public static bool IsDefaultEqualsOperator(IMethodBodyOperation operation)
         {
-            var body = GetBlockOfMethodBody(operation);
-            var bodyOps = body?.Operations ?? ImmutableArray<IOperation>.Empty;
             var _1 = ArrayBuilder<ArgumentSyntax>.GetInstance(out var initializerBuilder);
             var _2 = ArrayBuilder<SyntaxNode>.GetInstance(out var removalBuilder);
 
             // make sure the assignment wouldn't reference local variables we may have declared
-            var assignmentValues = GetAssignmentValuesForConstructor(bodyOps,
+            var assignmentValues = GetAssignmentValuesForConstructor(operation,
                 assignment => IsSafeAssignment(assignment)
                     ? assignment.Syntax as ExpressionSyntax
                     : null);
 
-            foreach (var property in positionalParams)
-            {
-                var assignmentIndex = assignmentValues.IndexOf(value => property.Equals(value.left));
-                if (assignmentIndex == -1)
+            AddArgumentAndRemovalAssignments(assignmentValues, positionalParams, initializerBuilder, removalBuilder);
+
+            return (initializerBuilder.AsImmutable(), removalBuilder.AsImmutable());
+        }
+
+        public static (ImmutableArray<ArgumentSyntax> initializerArguments,
+            ImmutableArray<AssignmentExpressionSyntax> assignmentsToRemove)
+        GetConstructorArgumentsFromObjectCreation(
+            ObjectCreationExpressionSyntax expressionNode,
+            ImmutableArray<IPropertySymbol> positionalParams,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            // we want to be very careful about when we refactor because if the user has a constructor
+            // and initializer it could be what they intend. Since we gave initializers to all non-primary
+            // constructors they already have, any calls to an explicit constructor with additional block initialization
+            // still work absolutely fine. Further, we can't necessarily associate their constructor args to
+            // primary constructor args or any other constructor args. Therefore,
+            // the only time we want to actually make a change is if they use the default no-param constructor,
+            // and a block initializer.
+            var operation = semanticModel.GetRequiredOperation(expressionNode, cancellationToken);
+            if (operation is IObjectCreationOperation
                 {
-                    // no assignment found, use "default" (or "null" for nullable types)
-                    initializerBuilder.Add(SyntaxFactory.Argument(SyntaxFactory.LiteralExpression(
-                        property.Type.NullableAnnotation == NullableAnnotation.Annotated
-                            ? SyntaxKind.NullLiteralExpression
-                            : SyntaxKind.DefaultLiteralExpression)));
-                }
-                else
+                    Arguments: ImmutableArray<IArgumentOperation> args,
```

Comment:
1. please don't abbreviate :)
2. if 'args' isn't used below, you can jsut do:  `Arguments.IsDefaultOrEmpty: true`
3. when are the arguments default?